dts/bindings/pinctrl/microchip,xec-pinctrl.yaml

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2020 Linaro Limited
+# Copyright (c) 2021 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    Microchip XEC Pin controller Node
+    Based on pincfg-node.yaml binding.
+
+    Note: `bias-disable` and `drive-push-pull` are default pin configurations.
+           They will be applied in case no `bias-foo` or `driver-bar` properties
+           are set.
+
+compatible: "microchip,xec-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node.yaml
+      child-binding:
+        property-allowlist:
+          - bias-disable
+          - bias-pull-down
+          - bias-pull-up
+          - drive-push-pull
+          - drive-open-drain
+          - output-high
+          - output-low
+
+properties:
+    reg:
+      required: true
+
+child-binding:
+    description: |
+        This binding gives a base representation of the Microchip XEC pins
+        configration
+
+    properties:
+        pinmux:
+            type: int
+            required: true
+            description: Pinmux selection
+
+        slew-rate:
+          required: false
+          type: string
+          default: "hw-default"
+          enum:
+          - "hw-default"        # Default value
+          - "low-speed"
+          - "high-speed"
+          description: Pin speed.
```

Comment:
what does "hw-default" mean?

Reply:
hw-default refers to the hardware default slew-rate config; for the device MEC172x, this hardware default is configurable through OTP option, hence we refer to that default here.

Reply:
Please, describe this hardware feature in detail, it's not clear from the bindings docs that these SoCs have this capability.

Reply:
![image](https://user-images.githubusercontent.com/25011557/152317113-b4df2541-6d37-491f-90d6-ead7207c02c8.png)
 seems like an edge case that should be documented here, too.

Reply:
will update the yaml doc to include following:

```
Some gpio pins do not implement all features. For example, the eSPI 
pins do not support drive strength or slew-rate configurability.
Please check datasheet to confirm. 
```